johnzon-jsonb/src/main/java/org/apache/johnzon/jsonb/JsonbAccessMode.java

```diff
@@ -868,8 +868,10 @@ public void close() throws IOException {
         ReaderConverters(final DecoratedType annotationHolder) {
             final boolean numberType = isNumberType(annotationHolder.getType());
             final boolean dateType = isDateType(annotationHolder.getType());
+            final boolean hasRawType = hasRawType(annotationHolder.getType());
             final JsonbTypeDeserializer deserializer = annotationHolder.getAnnotation(JsonbTypeDeserializer.class);
             final JsonbTypeAdapter adapter = annotationHolder.getAnnotation(JsonbTypeAdapter.class);
+            final JsonbTypeAdapter typeAdapter = hasRawType ? getRawType(annotationHolder.getType()).getDeclaredAnnotation(JsonbTypeAdapter.class) : null;
```

Comment:
Shouldnt it be done in the decorated type? There shouldnt be much type reflection there. Wdyt?

Reply:
You mean the hasRawType and getRawType stuff? I take a look into that

Reply:
Yep, today it is not super clean for the class (it is for fields and methods) but we support meta annotations and things like that, you can have a look to findFactory in the access mode and org.apache.johnzon.jsonb.JsonbAccessMode#getAnnotation. Guess you "just" want to reuse org.apache.johnzon.jsonb.JsonbAccessMode.ClassDecoratedType somehow here.